lib/src/connector/connection.cc

```diff
@@ -338,7 +338,11 @@ std::string const& Connection::getWsUri() {
 }
 
 void Connection::switchWsUri() {
+    auto old_t = connection_target_.load();
     ++connection_target_;
+    auto current_t = connection_target_.load();
```

Comment:
What I tried to say in the first comment was that you should use
```
auto old_t = getWsUri();
++connection_target_;
auto current_t = getWsUri();
```

connection_target_ is an integer; it could be renamed to clarify.